src/internal/signalr/handler.go

```diff
@@ -62,9 +74,53 @@ func NewObservationsHandler(thing adapter.Thing, cache cache.Cache) (Handler, er
 		model.LockCablePermanently:  handler.handleLockCablePermanently,
 	}
 
+	go handler.manageEnergyObservation()
+
 	return &handler, nil
 }
 
+func (h *observationsHandler) manageEnergyObservation() {
+	meterElecSrv, err := h.getMeterElecService()
+	if err != nil {
+		log.Error(err)
```

Comment:
If there's an error here, the function should return. I'd change the logic here and would pass a meter elec service as a parameter.